Firestore/Swift/Source/Codable/DocumentID.swift

```diff
@@ -101,9 +105,12 @@ public struct DocumentID<Value: DocumentIDWrappable & Codable>:
   // MARK: - `Codable` implementation.
 
   public init(from decoder: Decoder) throws {
-    throw FirestoreDecodingError.decodingIsNotSupported(
-      "DocumentID values can only be decoded with Firestore.Decoder"
-    )
+    guard let reference = decoder.userInfo[documentRefUserInfoKey] as? DocumentReference else {
+      throw FirestoreDecodingError.decodingIsNotSupported(
```

Comment:
With this change document ID decoding is no longer not supported, so the name of this error should change.

Reply:
@morganchen12 @andrewheard @mortenbekditlevsen @ncooke3 I'm going to resolve in order to merge today, but would appreciate a resolution/ follow up PR in the next few days

Reply:
`Firestore.Decoder` sets `userInfo[documentRefUserInfoKey]` in [`decode(_:from:in:)`](https://github.com/mortenbekditlevsen/firebase-ios-sdk/blob/c8535a5edf74bbbbb8aed3c9b9c4ded883395d86/Firestore/Swift/Source/Codable/EncoderDecoder.swift#L102-L104) so `DocumentID` decoding still works (as long as you use `Firestore.Decoder`, same as before).